cloudinit/config/schema.py

```diff
@@ -574,6 +576,100 @@ def validate_cloudconfig_metaschema(validator, schema: dict, throw=True):
         )
 
 
+def netplan_validate_network_config(
+    network_config_file: str,
+    strict: bool = False,
+    annotate: bool = False,
+    log_details: bool = True,
+) -> bool:
+    """On systems with python3-netplan, validate network_config schema for file
+
+    @param strict: Boolean, when True raise SchemaValidationErrors instead of
+       logging warnings.
+    @param log_details: Boolean, when True logs details of validation errors.
+       If there are concerns about logging sensitive userdata, this should
+       be set to False.
+
+    Leverage NetplanParser for error annotation line, column and detailed
+    errors.
+
+    When no network_config_file is provided, validate all netplan config files
+    in /etc/netplan which could have been delivered in the image outside of
+    cloud-init.
+
+    @return: True when schema validation was performed. False when not on a
+        system with netplan and netplan python support.
+    @raises: SchemaValidationError when netplan's parser raises
+        NetplanParserExceptions.
+    """
+    try:
+        from netplan import NetplanParserException, Parser
+    except ImportError:
+        LOG.debug(
+            "Skipping netplan schema validation. No python3-netplan available"
+        )
+        return False
+
+    # netplan Parser looks at all *.yaml files in the target directory under
+    # /etc/netplan. cloud-init should only validate schema of the
+    # network-config it generates, so create a <tmp_dir>/etc/netplan
+    # to validate only our network-config.
+    parse_dir = mkdtemp()
+    netplan_file = os.path.join(parse_dir, "etc/netplan/network-config.yaml")
```

Comment:
Unfortunately, yes it is. netplan Parser.load_yaml_hierarchy only looks under the subdir path etc/netplan/ to process yaml parts. Otherwise it happily ignores files in the root subdir of a path